ProcessLib/PhaseField/CreatePhaseFieldProcess.cpp

```diff
@@ -0,0 +1,201 @@
+namespace PhaseField
+{
+template <int DisplacementDim>
+class PhaseFieldProcess;
+
+extern template class PhaseFieldProcess<2>;
+extern template class PhaseFieldProcess<3>;
+
+template <int DisplacementDim>
+std::unique_ptr<Process> createPhaseFieldProcess(
+    MeshLib::Mesh& mesh,
+    std::unique_ptr<ProcessLib::AbstractJacobianAssembler>&& jacobian_assembler,
+    std::vector<ProcessVariable> const& variables,
+    std::vector<std::unique_ptr<ParameterBase>> const& parameters,
+    unsigned const integration_order,
+    BaseLib::ConfigTree const& config)
+{
+    //! \ogs_file_param{prj__processes__process__type}
+    config.checkConfigParameter("type", "PHASE_FIELD");
+    DBUG("Create PhaseFieldProcess.");
+
+    // Process variable.
+
+    //! \ogs_file_param{prj__processes__process__PHASE_FIELD__process_variables}
+    auto const pv_config = config.getConfigSubtree("process_variables");
+
+    auto process_variables = findProcessVariables(
+        variables, pv_config,
+        {//! \ogs_file_param_special{prj__processes__process__PHASE_FIELD__process_variables__phasefield}
+         "phasefield",
+         //! \ogs_file_param_special{prj__processes__process__PHASE_FIELD__process_variables__displacement}
+         "displacement"});
+
+    DBUG("Associate displacement with process variable \'%s\'.",
+         process_variables[1].get().getName().c_str());
+
+    if (process_variables[1].get().getNumberOfComponents() != DisplacementDim)
+    {
+        OGS_FATAL(
+            "Number of components of the process variable '%s' is different "
+            "from the displacement dimension: got %d, expected %d",
+            process_variables[1].get().getName().c_str(),
+            process_variables[1].get().getNumberOfComponents(),
+            DisplacementDim);
+    }
+
+    DBUG("Associate phase field with process variable \'%s\'.",
+         process_variables[0].get().getName().c_str());
+    if (process_variables[0].get().getNumberOfComponents() != 1)
+    {
+        OGS_FATAL(
+            "Phase field process variable '%s' is not a scalar variable but has "
+            "%d components.",
+            process_variables[0].get().getName().c_str(),
+            process_variables[0].get().getNumberOfComponents(),
+            DisplacementDim);
+    }
+
+    // Constitutive relation.
+    // read type;
+    auto const constitutive_relation_config =
+        //! \ogs_file_param{prj__processes__process__PHASE_FIELD__constitutive_relation}
+        config.getConfigSubtree("constitutive_relation");
+
+    auto const type =
+        //! \ogs_file_param{prj__processes__process__PHASE_FIELD__constitutive_relation__type}
+        constitutive_relation_config.peekConfigParameter<std::string>("type");
+
+    std::unique_ptr<MaterialLib::Solids::PhaseFieldExtension<DisplacementDim>>
+        material = nullptr;
+    if (type == "LinearElasticIsotropicPhaseField")
+    {
+        material = MaterialLib::Solids::createLinearElasticIsotropicPhaseField<
```

Comment:
You can use MaterialLib::Solids::createLinearElasticIsotropic here directly without modifying MaterialLib::Solids::createLinearElasticIsotropic.